pkg/kv/kvserver/kvflowcontrol/replica_rac2/processor.go

```diff
@@ -606,24 +613,34 @@ func (p *processorImpl) makeStateConsistentRaftMuLocked(
 	if termChanged {
 		p.term = state.Term
 	}
+	leadChanged := state.Leader != p.leaderID
+	leaseChanged := state.Leaseholder != p.leaseholderID
```

Comment:
@sumeerbhola Are we fine only checking the `leaseholderID`, or we should also check the lease epoch etc? Similarly to how we use `termChanged` here to disambiguate between "leadership has been contiguous" vs "we are still the leader but at a different term, so there could be another leader in between".